File: NTime/BaseCore/DataBase/Entities/Competition.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace BaseCore.DataBase
{
    public class Competition : IEntityId
    {
        public Competition()
        {
        }

        public Competition(string name, DateTime eventDate, string description, string link, string organizer, string city, string linkDisplayedName = "Link")
        {
            Name = name;
            EventDate = eventDate;
            Description = description;
            Link = link;
            Organizer = organizer;
            City = city;
            LinkDisplayedName = linkDisplayedName;
        }

        public int Id { get; set; }

        [Required]
        [StringLength(255)]
        public string Name { get; set; }

        public DateTime EventDate { get; set; }

        public DateTime? SignUpEndDate { get; set; }

        public string Description { get; set; }

        public string ExtraDataHeaders { get; set; }


        [StringLength(2000)]
        public string Link { get; set; }

        public string LinkDisplayedName { get; set; }

        [StringLength(255)]
        public string Organizer { get; set; }

        [StringLength(255)]
        public string City { get; set; }

        public bool OrganizerEditLock { get; set; }

        public virtual ICollection<Player> Players { get; set; }

        public virtual ICollection<Subcategory> Subcategories { get; set; }

        public virtual ICollection<Distance> Distances { get; set; }

        public virtual ICollection<AgeCategory> AgeCategories { get; set; }

        public virtual ICollection<OrganizerAccount> OrganizerAccounts { get; set; }

        public ICollection<ExtraColumn> ExtraColumns { get; set; }
    }
}

File: NTime/Server/ntimeAngular/src/app/Models/Competitions/Competition.ts
import { ExtraColumn } from '../ExtraColumns/ExtraColumn';
import { ICompetition } from './ICompetition';

export class Competition implements ICompetition {
  public Name: string;
  public ExtraDataHeaders: string;
  public Link: string;
  public LinkDisplayedName: string;
  public Organizer: string;
  public OrganizerEditLock: boolean;
  public ExtraColumns: ExtraColumn[];

  constructor(
    public Id: number,
    public City: string,
    public EventDate: Date,
    public SignUpEndDate: Date
  ) {}

  public static convertDates(competition: Competition): Competition {
    competition.EventDate = new Date(competition.EventDate);
    competition.SignUpEndDate = new Date(competition.SignUpEndDate);
    return competition;
  }

  copyDataFromDto(dto: ICompetition): void {
    this.Id = dto.Id;
    this.Name = dto.Name;
    this.ExtraDataHeaders = dto.ExtraDataHeaders;
    this.Link = dto.Link;
    this.LinkDisplayedName = dto.LinkDisplayedName;
    this.Organizer = dto.Organizer;
    this.OrganizerEditLock = dto.OrganizerEditLock;
    this.ExtraColumns = dto.ExtraColumns;
  }
}
